Src/Fido2/AttestationFormat/AppleAppAttest.cs

```diff
@@ -0,0 +1,124 @@
+                    {
+                        // App ID is the concatenation of your 10-digit team identifier, a period, and your app's CFBundleIdentifier value 
+                        s.CheckExactSequenceLength(1);
+                        s[0].CheckTag(Asn1Tag.PrimitiveOctetString);
+                        return s[0].GetOctetString();
+                    }
+                }
+            throw new Fido2VerificationException("Apple AppAttest attestation extension 1.2.840.113635.100.8.5 has invalid data");
+        }
+
+        // From https://www.apple.com/certificateauthority/Apple_App_Attestation_Root_CA.pem
+        internal static readonly string appleAppAttestationRootCA = "MIICITCCAaegAwIBAgIQC/O+DvHN0uD7jG5yH2IXmDAKBggqhkjOPQQDAzBSMSYwJAYDVQQDDB1BcHBsZSBBcHAgQXR0ZXN0YXRpb24gUm9vdCBDQTETMBEGA1UECgwKQXBwbGUgSW5jLjETMBEGA1UECAwKQ2FsaWZvcm5pYTAeFw0yMDAzMTgxODMyNTNaFw00NTAzMTUwMDAwMDBaMFIxJjAkBgNVBAMMHUFwcGxlIEFwcCBBdHRlc3RhdGlvbiBSb290IENBMRMwEQYDVQQKDApBcHBsZSBJbmMuMRMwEQYDVQQIDApDYWxpZm9ybmlhMHYwEAYHKoZIzj0CAQYFK4EEACIDYgAERTHhmLW07ATaFQIEVwTtT4dyctdhNbJhFs/Ii2FdCgAHGbpphY3+d8qjuDngIN3WVhQUBHAoMeQ/cLiP1sOUtgjqK9auYen1mMEvRq9Sk3Jm5X8U62H+xTD3FE9TgS41o0IwQDAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBSskRBTM72+aEH/pwyp5frq5eWKoTAOBgNVHQ8BAf8EBAMCAQYwCgYIKoZIzj0EAwMDaAAwZQIwQgFGnByvsiVbpTKwSga0kP0e8EeDS4+sQmTvb7vn53O5+FRXgeLhpJ06ysC5PrOyAjEAp5U4xDgEgllF7En3VcE3iexZZtKeYnpqtijVoyFraWVIyd/dganmrduC1bmTBGwD";
+
+        public static readonly X509Certificate2 AppleAppAttestRootCA = new(Convert.FromBase64String(appleAppAttestationRootCA));
+
+        // From https://developer.apple.com/documentation/devicecheck/validating_apps_that_connect_to_your_server
+        // "aaguid field is either appattestdevelop if operating in the development environment..."
+        // 61707061-7474-6573-7464-6576656c6f70
+        public static readonly Guid devAaguid = new("61707061-7474-6573-7464-6576656c6f70");
+
+        // "...or appattest followed by seven 0x00 bytes if operating in the production environment"
+        // 61707061-7474-6573-7400-000000000000
+        public static readonly Guid prodAaguid = new("61707061-7474-6573-7400-000000000000");
+
+        public override (AttestationType, X509Certificate2[]) Verify()
+        {
+            // 1. Verify that the x5c array contains the intermediate and leaf certificates for App Attest, starting from the credential certificate in the first data buffer in the array (credcert).
+            if (!(X5c is CborArray { Length: 2 } x5cArray && x5cArray[0] is CborByteString { Length: > 0 } && x5cArray[1] is CborByteString { Length: > 0 }))
+            {
+                throw new Fido2VerificationException("Malformed x5c in Apple AppAttest attestation");
+            }
+
+            // Verify the validity of the certificates using Apple's App Attest root certificate.
+            var chain = new X509Chain();
+            chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
```

Comment:
Maybe: NoRevocationCheck when in dev environment?

Reply:
In the samples I have seen for both dev and prod there are no CDP extensions present, so no way to check CRL regardless.